### PR TITLE
Rework Flux2 website intro

### PR DESCRIPTION
**Pull request by squaremo**
- adds a section on target users
 - moves GitOps Toolkit info into its own index
 - gives a brief account of how you use Flux
 - clarifies (or so I claim) some of the target features

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -10,62 +10,68 @@ components of the Kubernetes ecosystem. In version 2, Flux supports
...
 !!!hint "Get started with Flux v2!"
     Following this [guide](get-started/index.md) will just take a couple of minutes to complete:
     After installing the `flux` CLI and running a couple of very simple commands,
     you will have a GitOps workflow setup which involves a staging and a production cluster.
 
+## More detail on what's in Flux
+
+Target features:
+
+- Source configuration from Git and S3 (and Minio)
```
Comment (stefanprodan) [inline docs/index.md:61]:
```suggestion
- Source configuration from Git and Helm repositories, S3 compatible storage like Minio
```

File: docs/dev-guides/index.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# GitOps Toolkit
```
Comment (stefanprodan) [inline docs/dev-guides/index.md:1]: I think this should be the index of "Toolkit Components" and linked in the menu as "Overview".

**Comment by dholbach** [comment]
You might want to add `docs/dev-guides/index.md` to `mkdocs.yml`.

File: docs/dev-guides/index.md (hunk 2 above)
Comment (squaremo) [reply docs/dev-guides/index.md:1]: Yarp OK

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @squaremo 🍰 

PS: please get the talks section from main branch

**Comment by squaremo** [comment]
> PS: please get the talks section from main branch

Sure -- should we avoid duplication with fluxcd/community somehow?

**Comment by stefanprodan** [comment]
> should we avoid duplication with fluxcd/community somehow?

I wanted to do that but @dholbach argued that's better to keep the list everywhere for better visibility. Let's keep it like this for now.

**Comment by dholbach** [comment]
Tamao wanted this. Once we have the new website, we should make sure the talk information is very visible and just have it in one place there.

Then we can remove it from the individual repos.

**Comment by squaremo** [comment]
ugh, nightmare rebass ☠️ 🐟

**Comment by dholbach** [comment]
:bouquet:

**Pull request merged**